ChiTech/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_01c_finalize.cc

```diff
@@ -3,256 +3,272 @@
+          throw std::runtime_error(
+              "Total fission yield must be greater than unity.");
 
-    //============================== Compute total from prompt and delayed
-    if (has_nu_p and has_nu_d)
-    {
-      for (int g = 0; g < num_groups; ++g)
-        nu[g] = nu_prompt[g] + nu_delayed[g];
-      has_nu = true;
-    }
+        //compute other quantities
+        if (was_specified(nu_sigma_f))
+        {
+          for (unsigned int g = 0; g < num_groups; ++g)
+            sigma_f[g] = nu[g] * sigma_f[g];
+        }
+        else if (was_specified(nu_sigma_f))
+        {
+          for (unsigned int g = 0; g < num_groups; ++g)
+            if (nu[g] != 0.0)
+              sigma_f[g] = nu_sigma_f[g] / nu[g];
+        }
+        else
+          throw std::runtime_error(
+              "Neither the fission cross-section nor the "
+              "fission multiplicity cross-section was specified.");
 
-    if (has_chi_p and has_chi_d)
-    {
-      for (int g = 0; g < num_groups; ++g)
-      {
-        double beta = nu_delayed[g] / nu[g];
-        chi[g] = (1.0 - beta) * chi_prompt[g];
-        for (int j = 0; j < num_precursors; ++j)
-          chi[g] += beta * precursor_yield[j] * chi_delayed[g][j];
-      }
-      has_chi = true;
-    }
+        //check that chi was specified
+        if (!was_specified(chi))
+          throw std::runtime_error(
+              "Total fission spectrum was not provided.");
 
-    //======================================== Make sure appropriate quantities
-    //                                         were provided
-    if (num_precursors > 0)
-    {
-      if (not has_nu_p or not has_nu_d)
-      {
-        chi::log.Log0Error()
-          << __FUNCTION__ << ": Both prompt and delayed nu must be provided "
-          << "when precursors are present.";
-        chi::Exit(EXIT_FAILURE);
-      }
-      if (not has_chi_p or not has_chi_d)
-      {
-        chi::log.Log0Error()
-            << __FUNCTION__ << ": Both prompt and delayed chi must be "
-            << "provided when precursors are present.";
-        chi::Exit(EXIT_FAILURE);
+        //normalize the total fission spectrum
+        double chi_sum = std::accumulate(chi.begin(), chi.end(), 0.0);
+        for (unsigned int g = 0; g < num_groups; ++g)
+          chi[g] /= chi_sum;
       }
     }
-    else
+    catch (const std::runtime_error& err)
     {
-      if (not has_nu)
-      {
-        chi::log.Log0Error()
-            << __FUNCTION__ << ": Total nu must be provided "
-            << "when precursors are present.";
-        chi::Exit(EXIT_FAILURE);
-      }
-      if (not has_chi)
-      {
-        chi::log.Log0Error()
-            << __FUNCTION__ << ": Total chi must be "
-            << "provided when precursors are present.";
-        chi::Exit(EXIT_FAILURE);
-      }
+      chi::log.LogAllError()
+          << "Invalid fission data encountered.\n"
+          << "Error Message:\n" << err.what();
+      chi::Exit(EXIT_FAILURE);
```

Comment:
we have different way of exiting when an error is found. here, you use chi::Exit() but earlier, you used std::runtime_error(). Do they have the same functionality? Should we stick to one?

Reply:
All of those `throw std::runtime_error()` are inside of a try block and the code here is the catch block. The utility of this is that I don't have to both print the error message and exit at every check. When one of the runtime errors get thrown, the code you have highlighted here gets executed and `chi::Exit` gets called. The utility of this is that I don't have to have curly braces around the million checks there are!

Reply:
Created issue #271 per discussion with @Naktakala